jupyterhub/schema.yaml

```diff
@@ -467,6 +467,28 @@ properties:
   proxy:
     type: object
     properties:
+      chp:
+        type: object
+        description: |
+          Configure the configurable-http-proxy (chp) pod managed by jupyterhub to route traffic
+          both to itself and to user pods.
+        properties:
+          extraEnv:
+            type: array
+            description: |
+              Provide extra environment variables to the chp pod. Can be provided as an array or
+              as an object.
+              
+              as an array:
+              ```yaml
+                - name: MY_ENVVAR
+                  value: MY_ENVVAR_VALUE
+              ```
+
+              as an object:
+              ```yaml
+                MY_ENVVAR: MY_ENVVAR_VALUE
+              ```
```

Comment:
Thank you for excellent work and documenting these in schema.yaml!

I'm a bit worried that the examples provided may indicate that its the only way to specify something, which could exclude the ability to define an environment variable in a k8s native syntax for example. The examples in `hub.extraEnv` and `singleuser.extraEnv` are more complete, perhaps you could copy paste from them, reference them, or extend your examples to cover the various options?

Feel free to update those examples as well if you like to change something for cohesiveness.

Thank you for your work on this @agrahamlincoln !!